service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java

```diff
@@ -41,6 +41,13 @@ public Instant checkPermission(String accessToken) {
         samClient.setAccessToken(accessToken);
         var resourceApi = new ResourcesApi(samClient);
 
+        // check that user has access to workspace
+        boolean workspaceAccess = resourceApi.resourcePermissionV2("workspace", samResourceId, "read");
```

Comment:
I think `samResourceId` is wrong -- that would be the ID of the runtime/kubernetes-app, not the workspace.

You might need to add a new contructor-arg to this class and pass it from AppConfiguration (which should have the the workspaceId for other things).

Reply:
Ah I see. It looks like `setDateAccessedInspectorProperties` in application config has workspace ID. I will use that to pass it to SamResourceClient. Thank you for pointing this out!